static_analyzer/engine/source_inspector.py

```
@@ -26,178 +97,210 @@ def get_file_lines(self, file_path: Path) -> list[str] | None:
+            if pos in seen:
+                continue
+            seen.add(pos)
+            sites.append(CallSite(file=str(file_path), line=pos[0] + 1, column=pos[1] + 1))
+        return sites
+
+    def _read_file_bytes(self, file_path: Path) -> bytes | None:
+        file_key = str(file_path)
+        if file_key not in self._file_bytes_cache:
+            try:
+                self._file_bytes_cache[file_key] = file_path.read_bytes()
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)